### DISCUSSION TITLE
Upgrade path from 1.0 datamodel

### DISCUSSION THREAD

**Discussion by clioh** [prisma/prisma#2225] [Q&A] [answered]
I posted this question in the slack channel but I didn't get a response, so I was hoping if I reposted here, I might be able to resolve my problem.I'm excited to try the new product as I've been on Prisma for a while now and have a pretty sizable database on it. I started poking around Prisma 2 finally today and it has a lot of features I'm really excited about. I used `prisma introspect` to lift the schema out of our existing database, but I'm having a little trouble with @relation directive that I was hoping to get some insight on. I read through the docs here but I'm not sure I followed. From my schema definition, I have:
```
type User {
  accounts: [Account!] @relation(name: "UserAccounts", onDelete: CASCADE)
  ...
}

type Account {
  user: User! @relation(name: "UserAccounts")
}
```

We started prior to the release of the 1.1 datamodel and as such, we have relation tables in the database (`_UserAccounts`, etc), but no foreign key relation defined in either table, so we're sort of stuck here. My understanding is that prisma 2 is only going to support foreign key relations for 1:n relationships, which means that we'll have to alter our database schema to support these. Is that correct?

In doing some research, I came across this the guide [here](https://v1.prisma.io/docs/1.31/releases-and-maintenance/features-in-preview/datamodel-v11-b6a7/#4.-optimizing-the-database-schema) that explains how to upgrade from `1.0` to `1.1`. I followed it and `prisma export`'d our dev database, made the changes to support the `INLINE` strategy, but the foreign key relations didn't populate. Digging through the CLI code, it doesn't look like that's a feature supported by `prisma import`.

I'm wondering what our upgrade path is here. Do we need to do a manual migration to add foreign keys to all of our tables where relationships are defined by a relation table?

Thanks in advance,
Clio

**Comment by nikolasburk** [comment, marked as answer]
Hey @harpe116 👋 apologies for the delay on this one!

> My understanding is that prisma 2 is only going to support foreign key relations for 1:n relationships, which means that we'll have to alter our database schema to support these. Is that correct?

Indeed, all relations in Prisma 2 are always based on foreign keys in the underlying DB.

> I'm wondering what our upgrade path is here. Do we need to do a manual migration to add foreign keys to all of our tables where relationships are defined by a relation table?

We are currently in the process of researching and defining the best upgrade paths for Prisma 1 users and will share more about this soon. You can track the developments in this GitHub issue where we're posting updates about the research: https://github.com/prisma/prisma/issues/1937

It'll likely be the case that some schema wrangling with SQL will be necessary for a proper upgrade – our goal is to build additional tools to help you with the upgrade (e.g. by auto-generating the SQL commands that you need to run to make the DB schema compatible with Prisma 2). 

So, unfortunately I don't have a perfect answer for you right now (other than asking for a bit more patience until we have figured things out internally and then share our findings publicly 🙏 ).

Please feel free to ping me directly on the Prisma Slack (I'm `@nikolasburk` there, I'd be happy to help you figure things out there)!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
